### PR TITLE
20778 Update business get response to return alternate names list

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#20778

*Description of changes:*
- Updated `json` function in business model to return `alternateNames` list when enable-legal-name-fix feature flag is on. 
- Get business endpoint returns `alternateNames`
- Added unit tests to verify alternate names are being returned correctly

**Businesses tested**
```
Using: GET <api_base_url>/api/v2/businesses/<identifier>

// COOP
"CP1002603",
"alternateNames": [ ]

// BEN - no DBAs, with translation
"BC0871269"
"alternateNames": [
         {
            "name": "IUIOUI",
            "startDate": "2020-12-05",
            "type": "TRANSLATION"
        }
]

// BEN - multiple DBAs, with translation
"BC1230010"
"alternateNames": [
        {
            "name": "LOREM IPSUM",
            "startDate": "2020-10-28",
            "type": "TRANSLATION"
        },
        {
            "entityType": "SP",
            "identifier": "FM1032630",
            "name": "FIX",
            "registeredDate": "2023-03-16T21:35:43.599246+00:00",
            "startDate": "2023-03-15",
            "type": "DBA"
        },
        {
            "entityType": "SP",
            "identifier": "FM1032736",
            "name": "SOME DBA TEST",
            "registeredDate": "2023-03-17T23:39:25.980718+00:00",
            "startDate": "2023-03-16",
            "type": "DBA"
        }
],

// SP - individual
"FM1028594"
"alternateNames": [
        {
            "entityType": "SP",
            "identifier": "FM1028594",
            "name": "AC SP 2023.FEB.01 21.13 TEST",
            "registeredDate": "2023-02-02T23:26:11.664620+00:00",
            "startDate": "2022-10-04",
            "type": "DBA"
        }
],


// SP - LEAR DBA
"FM1054864"
"alternateNames": [
        {
            "entityType": "SP",
            "identifier": "FM1054864",
            "name": "AC SP DBA 2023.SEP.29 14.46 TEST",
            "registeredDate": "2023-10-03T14:10:58.360585+00:00",
            "startDate": "2023-09-07",
            "type": "DBA"
        }
],

// SP - COLIN DBA
"FM1061225"
"alternateNames": [
        {
            "entityType": "SP",
            "identifier": "FM1061225",
            "name": "AC SP DBA 2023.DEC.08 22.20 TEST",
            "registeredDate": "2023-12-08T22:48:40.954878+00:00",
            "startDate": "2023-12-08",
            "type": "DBA"
        }
],

// GP - 2 partners - 2 persons
"FM1018902"
"alternateNames": [
        {
            "entityType": "GP",
            "identifier": "FM1018902",
            "name": "AC GP 2022.OCT.25 21.21 TEST",
            "registeredDate": "2022-10-25T21:38:02.636697+00:00",
            "startDate": "2022-12-04",
            "type": "DBA"
        }
],

// GP - 2 partners - 1 person + 1 LEAR org, 1 DBA
"FM1044962"
"alternateNames": [
        {
            "entityType": "SP",
            "identifier": "FM1054479",
            "name": "AC SP DBA 2023.SEP.29 14.02 TEST",
            "registeredDate": "2023-09-29T14:21:59.059242+00:00",
            "startDate": "2023-09-27",
            "type": "DBA"
        },
        {
            "entityType": "GP",
            "identifier": "FM1044962",
            "name": "AC GP 2023.JUN.22 13.29 TEST",
            "registeredDate": "2023-06-22T14:33:25.133300+00:00",
            "startDate": "2023-06-22",
            "type": "DBA"
        }
],

// GP - 2 partners - 1 person + 1 COLIN org
"FM1000001"
"alternateNames": [
        {
            "entityType": "GP",
            "identifier": "FM1000001",
            "name": "INDUSTIRES GP",
            "registeredDate": "2022-02-01T00:00:00+00:00",
            "startDate": "2020-08-24",
            "type": "DBA"
        }
],
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/models/business.py [hunk 1]
```diff
@@ -561,6 +571,68 @@ def get_filing_by_id(cls, business_identifier: int, filing_id: str):
...
+                            'entityType': sole_prop.legal_type,
+                            'identifier': sole_prop.identifier,
+                            'name': sole_prop.legal_name,
+                            'registeredDate': sole_prop.founding_date.isoformat(),
+                            'startDate': start_date,
+                            'type': 'DBA'
+                        }
+                    )
+
+        if self.legal_type in [Business.LegalTypes.SOLE_PROP, Business.LegalTypes.PARTNERSHIP]:
```
Comment (chenhongjing) [inline legal-api/src/legal_api/models/business.py:621]: We can use `self.is_firm` here. 😄

File: legal-api/tests/unit/models/test_business.py [hunk 2]
```diff
@@ -305,6 +306,10 @@ def test_business_json(session):
         'allowedActions': {}
     }
 
+    from legal_api.services import flags  # pylint: disable=import-outside-toplevel
+    if flags.is_on("enable-legal-name-fix"):
```
Comment (chenhongjing) [inline legal-api/tests/unit/models/test_business.py:310]: Can we mock flag on&off values here?

File: legal-api/src/legal_api/models/business.py [hunk 3]
```diff
@@ -561,6 +571,68 @@ def get_filing_by_id(cls, business_identifier: int, filing_id: str):
...
+        - For SP or GP, also return an alternate name from existing business record
+        - Return empty list if there are no alternate name entries
+        """
+        alternate_names = []
+
+        for alias in self.aliases:
+            alternate_names.append(
+                {
+                    'name': alias.alias,
+                    'startDate': LegislationDatetime.format_as_legislation_date(self.founding_date),
```
Comment (argush3) [inline legal-api/src/legal_api/models/business.py:588]: I thought `aliases` had a start date field for some reason.

We'll need to get the start date value from filing.effective_date that's tied to the transaction that created the alias entry.

Included an example where there is more than one versioning record for a given alias entry.  But basically we just need to check that the `end_transaction_id` is null to ensure we are looking at the right transaction & filing.

image

File: legal-api/src/legal_api/models/business.py [hunk 4]
```diff
@@ -425,8 +430,13 @@ def _slim_json(self):
 
     def _extend_json(self, d):
         """Include conditional fields to json."""
+        from legal_api.services import flags  # pylint: disable=import-outside-toplevel
+
         base_url = current_app.config.get('LEGAL_API_BASE_URL')
 
+        if flags.is_on('enable-legal-name-fix'):
```
Comment (JazzarKarim) [inline legal-api/src/legal_api/models/business.py:437]: So this is how you use FFs in the BE. Simple and straightforward. Nice!

**Review by JazzarKarim — APPROVED** [review]
From what I see, LGTM 👍

File: legal-api/src/legal_api/models/business.py (hunk 3 above)
Comment (leodube-aot) [reply legal-api/src/legal_api/models/business.py]: I've updated the start date to grab it from the filing associated with the alias entry creation

**Review by argush3 — APPROVED** [review]

**Pull request merged**